python/cuml/tests/test_device_selection.py

```diff
@@ -20,13 +20,36 @@
 import numpy as np
 import pandas as pd
 import cudf
-from sklearn.datasets import make_regression
-from sklearn.linear_model import LinearRegression as skLinearRegression
-from cuml.linear_model import LinearRegression
+import pickle
+import inspect
+from importlib import import_module
+
+from pytest_cases import fixture_union, pytest_fixture_plus
+from sklearn.datasets import make_regression, make_blobs
+from cuml.metrics import trustworthiness
+
 from cuml.testing.test_preproc_utils import to_output_type
 from cuml.common.device_selection import DeviceType, using_device_type
 from cuml.common.memory_utils import MemoryType, using_memory_type
-import pickle
+
+from sklearn.linear_model import LinearRegression as skLinearRegression
+from cuml.linear_model import LinearRegression
+from sklearn.linear_model import LogisticRegression as skLogisticRegression
+from cuml.linear_model import LogisticRegression
```

Comment:
Is there a reason we're testing all these algorithms here as opposed to in their own test files? I think it's reasonable to have one algorithm in the device_selection tests, but I'm not sure what the benefit is of testing more. We certainly don't want to have all algorithms here in the long run.

Reply:
Sure, we can move the tests of algorithms to their respective test files. However, there's definitely a need for a test of the patterns of CPU/GPU training followed by CPU/GPU inference for each of the estimators to be able to detect the very many potential bugs that can emerge in different scenarios. That's why I thought it wasn't a bad idea to have the pattern as tests and a fixture for every of the models gathered rigorously in a single file as to not forget any.

Reply:
Let's leave this for a follow-on, but in general, the concept of "CPU/GPU interoperability" is not one that will stick around as an organizing principle for the codebase. Eventually, it will just be "here's an algorithm" and it will have both implementations.